ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -25,25 +72,17 @@ public final class Atomic<Value> {
 	}
 	
 	/// Initializes the variable with the given initial value.
-	public init(_ value: Value) {
+	public init(_ value: Value, mutex: MutexType = PosixThreadMutex()) {
```

Comment:
`lock`, `unlock`, `modify` and `withValue` routines where implemented twice, in `MutableProperty` and `Atomic`. `MutableProperty` knows that she wants `atomicity` built on `recursive` locking strategy, but implementation of these requirements can be delegated.